node/src/components/linear_chain.rs

```diff
@@ -122,7 +133,24 @@ where
                 responder.respond(self.last_block.clone()).ignore()
             }
             Event::Request(LinearChainRequest::BlockAtHeight(height, sender)) => {
-                let block_at_height = self.linear_chain.get(height as usize).map(|block| BlockByHeight::new(block.clone())).unwrap_or_else(|| BlockByHeight::Absent(height));
+                // Treat `linear_chain` as a cache of least-recently asked for blocks.
+                // match self.linear_chain.get(height as usize).cloned() {
+                //     Some(block) => effect_builder
+                //         .immediately()
+                //         .event(move |_| Event::GetBlockByHeightResult(height, Some(block), sender)),
+                //     None => 
```

Comment:
I want to keep this comment for reference, so that we don't forget that `self.linear_chain` should be kept as a cache.